pycbc/waveform/bank.py

```diff
@@ -394,24 +403,157 @@ def approximant(self, index):
     def __len__(self):
         return len(self.table)
 
-    def template_thinning(self, injection_parameters, threshold):
-        from pycbc.pnutils import mass1_mass2_to_tau0_tau3
+    def template_thinning(self, gwstrain):
+        if gwstrain.injcutter is None or \
+                gwstrain.injcutter['chirp_time_threshold'] is None:
```

Comment:
I don't like that this is reading a secret parameter. Sometimes it's necessary but I think here it should be given as an argument to the method.